tools/python_tools/cuvslam_tools/dataset_preparation/rgbd.py

```
@@ -0,0 +1,484 @@
+
+    The reporter compares odometry against a trajectory that starts at the
+    origin, so the first row is exactly the identity and later rows are
+    ``pose(frame 0)^-1 * pose(frame i)``.
+    """
+    timestamps = [row[0] for row in trajectory]
+    first_rotation, first_translation = interpolate_pose(trajectory, timestamps, pairs[0][0])
+    inverse_rotation, inverse_translation = invert_transform(first_rotation, first_translation)
+    lines = [
+        "1.000000e+00 0.000000e+00 0.000000e+00 0.000000e+00 "
+        "0.000000e+00 1.000000e+00 0.000000e+00 0.000000e+00 "
+        "0.000000e+00 0.000000e+00 1.000000e+00 0.000000e+00"
```

Ruff (0.16.2)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)